napari/plugins/_npe2.py

```diff
@@ -329,7 +329,11 @@ def on_plugins_registered(manifests: set[PluginManifest]):
 
     'Registered' means that a manifest has been provided or discovered.
     """
-    for mf in manifests:
+    sorted_manifests = sorted(
+        manifests,
+        key=lambda mf: mf.display_name if mf.display_name else mf.name,
```

Comment:
We may provide here better key function. 

```python
def _get_plugin_sort_key(mf: PluginManifest) -> str:
    if len(mf.contributions.widgets) == 1:
        return mf.contributions.widgets[0].display_name
    if mf.display_name:
        return mf.display_name
    return mf.name
```

This will solve the problem of ordering single widget contribution.

Reply:
Omg duh 😂 thanks @Czaki will fix up

Reply:
Hmm actually this will solve the Plugins menu case, but then Samples menu may be incorrectly ordered, right? I don't think there's a way we can get both correct at this stage in the code. We'd have to collect all contributions and sort before registering the actions with app-model.

Reply:
I think I know how to do it with minimal intervention actually. I'll follow up. If this PR gets merged beforehand that's fine

Reply:
I think that, the best solution, will be to improve `build_qmodel_menu` and `QModelMenu` For get dict of `id -> sorting function`

Reply:
Ok I'd say let's merge this PR as is, and we'll follow up for the single-contribution case